mycroft/util/log.py

```diff
@@ -51,99 +50,49 @@ def method(cls, *args, **kwargs):
-        cls.handler.setFormatter(formatter)
-
-        # Enable logging in external modules
-        cls.create_logger('').setLevel(cls.level)
-
-    @classmethod
-    def create_logger(cls, name):
-        logger = logging.getLogger(name)
-        logger.propagate = False
-        logger.addHandler(cls.handler)
-        return logger
-
-    def __init__(self, name):
-        LOG._custom_name = name
-
-    @classmethod
-    def _log(cls, func, *args, **kwargs):
-        if cls._custom_name is not None:
-            name = cls._custom_name
-            cls._custom_name = None
-        else:
-            # Stack:
-            # [0] - _log()
-            # [1] - debug(), info(), warning(), or error()
-            # [2] - caller
-            try:
-                stack = inspect.stack()
-
-                # Record:
-                # [0] - frame object
-                # [1] - filename
-                # [2] - line number
-                # [3] - function
-                # ...
-                record = stack[2]
-                mod = inspect.getmodule(record[0])
-                module_name = mod.__name__ if mod else ''
-                name = module_name + ':' + record[3] + ':' + str(record[2])
-            except Exception:
-                # The location couldn't be determined
-                name = 'Mycroft'
-
-        func(cls.create_logger(name), *args, **kwargs)
-
-
-LOG.init()
+LOG = logging.getLogger('Mycroft')
+
+
+def create_logger(name):
+    log_message_format = ('{asctime} | {levelname:8} | {process:5} | '
+                          '{name} | {message}')
+    handler = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(log_message_format, style='{')
+    formatter.default_msec_format = '%s.%03d'
+    handler.setFormatter(formatter)
+
+    logger = LOG.getChild(name)
+    logger.addHandler(handler)
+    logger.propagate = False
+    logger.addHandler(handler)
+    return logger
+
+
+def setup_default_logger(level='INFO'):
+    """ Setup the default logger."""
+    log_message_format = ('{asctime} | {levelname:8} | {process:5} | '
+                          '{module}:{lineno} | {message}')
+    handler = logging.StreamHandler(sys.stdout)
+    formatter = logging.Formatter(log_message_format, style='{')
+    formatter.default_msec_format = '%s.%03d'
+    handler.setFormatter(formatter)
+    LOG.addHandler(handler)
+
+    # Check configs manually, the Mycroft configuration system can't be
+    # used since it uses the LOG system and would cause horrible cyclic
+    # dependencies.
+    confs = [SYSTEM_CONFIG, USER_CONFIG]
+    config = {}
+    for conf in confs:
+        try:
+            merge_dict(config,
+                       load_commented_json(conf) if isfile(conf) else {})
+        except Exception as e:
+            print('couldn\'t load {}: {}'.format(conf, str(e)))
+    LOG.setLevel(logging.getLevelName(config.get('log_level', level)))
+
+    # Backwards compatibility, remove in rekease 20.02
```

Comment:
maybe add a TODO to the comment so we can easily grep the codebase and avoid missing this, seems to be the standard so far for stuff that will be deprecated in follow up versions

Reply:
Good thinking. I'll add that

Reply:
PR updated